coalib/core/Core.py

```diff
@@ -327,34 +149,176 @@ def instantiate_and_track(prev_bear_type, next_bear_type):
     return dependency_tracker, bears
 
 
-def run(bears, result_callback):
+class Session:
     """
-    Runs a coala session.
-
-    :param bears:
-        The bear instances to run.
-    :param result_callback:
-        A callback function which is called when results are available. Must
-        have following signature::
+    Maintains a session for a coala execution. For each session, there are set
+    of bears to run along with a callback function, which is run when results
```

Comment:
ah, we should substitute this:

"which is ~~run~~ called when results are available."

Avoiding using `run` twice.